src/react/ui-elements/Veeam/ VeeamCapacityModal.tsx

```diff
@@ -0,0 +1,143 @@
+  VEEAM_XML_PREFIX,
+  unitChoices,
+} from './VeeamConstants';
+import { useQueryClient } from 'react-query';
+
+type VeeamCapacityModalProps = {
+  isOpen: boolean;
+  close: () => void;
+  bucketName: string;
+  maxCapacity: string;
+};
+
+type Units = 'GiB' | 'TiB' | 'PiB';
+
+export const VeeamCapacityModalInternal = ({
+  isOpen,
+  close,
+  bucketName,
+  maxCapacity,
+}: VeeamCapacityModalProps) => {
+  const [capacity, setCapacity] = useState<string>(maxCapacity.split(' ')[0]);
+  const [capacityUnit, setCapacityUnit] = useState<string>(
+    `${unitChoices[maxCapacity.split(' ')[1] as Units]}`,
+  );
+  const capacityBytes = (
+    parseInt(capacity, 10) * parseInt(capacityUnit, 10)
+  ).toString();
+  const { mutate } = usePutObjectMutation();
+  const { showToast } = useToast();
+  const queryClient = useQueryClient();
+
+  return (
+    <Modal
+      close={close}
+      isOpen={isOpen}
+      footer={
+        <Wrap>
+          <p></p>
+          <Stack>
+            <Button variant="outline" onClick={close} label="Cancel" />
+            <Button
+              variant="primary"
+              onClick={() =>
+                mutate(
+                  {
+                    Bucket: bucketName,
+                    Key: `${VEEAM_XML_PREFIX}/capacity.xml`,
+                    Body: GET_CAPACITY_XML_CONTENT(capacityBytes),
+                    ContentType: 'text/xml',
+                  },
+                  {
+                    onSuccess: () => {
+                      close();
+                      showToast({
+                        open: true,
+                        status: 'success',
+                        message: 'Repository capacity updated successfully',
+                      });
+                      queryClient.invalidateQueries([
+                        'getVeeamObject',
+                        bucketName,
+                      ]);
+                    },
+                    onError: (error) => {
+                      showToast({
+                        open: true,
+                        status: 'error',
+                        message: `Failed to update repository capacity: ${error}`,
+                      });
+                    },
+                  },
+                )
+              }
+              label="Edit"
+              icon={<Icon name="Pencil" />}
+            />
+          </Stack>
+        </Wrap>
+      }
+      title="Edit max repository capacity"
+    >
+      <div
+        style={{
+          height: '15vh',
+        }}
+      >
+        <Stack direction={'vertical'} gap={'r4'}>
+          <Stack direction="horizontal">
+            <Input
+              id="capacity"
+              type="number"
+              // @ts-expect-error - TODO: Fix the type of the size props in Input component
+              size="1/3"
+              min={1}
+              max={999}
+              value={capacity}
+              onChange={({ target: { value } }) => setCapacity(value)}
+            />
```

Comment:
I would auto focus this input when the modal get's opened.

Reply:
Fxied in ab4750b6dd4acfb8df4454d716fd24a0a26074eb